src/core/buffers/orchestrator/resolve_partial_period.ts

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  merge as observableMerge,
+  Observable,
+  of as observableOf,
+} from "rxjs";
+import {
+  filter,
+  mergeMap,
+  take,
+} from "rxjs/operators";
+import { MediaError } from "../../../errors";
+import Manifest, {
+  LoadedPeriod,
+  PartialPeriod,
+} from "../../../manifest";
+import { fromEvent } from "../../../utils/event_emitter";
+
+export interface INeedsLoadedPeriodEvent { type : "needs-loaded-period";
+                                           value: { period: PartialPeriod }; }
+export interface IPeriodLoadedEvent { type : "period-loaded";
+                                      value : { period : LoadedPeriod }; }
+
+/**
+ * Try to obtain a LoadedPeriod from the PartialPeriod and wanted time given.
+ * @param {Object} Manifest
+ * @param {Object} period
+ * @param {number} wantedTime
+ * @returns {Observable}
+ */
+export default function resolvePartialPeriod(
```

Comment:
Not so clear the fact that you send an event to trigger period loading, then receive the event loadedPeriod. Isn't it risky ?

Reply:
Yeah it is. That's why I put it at first place in the `observableMerge` below (I don't know if there is another way to make sure the event listener is registered first).

I first did a ReplaySubject in PartialPeriod that would emit the loaded Period(s) but it was complex to trigger from the `Manifest` object, which had to call it at a moment when that period was loaded. 

At the end, the `Manifest` is already an event emitter and it seemed a good way to do it.

How would you do it?

Reply:
Oh perhaps you're speaking about the whole round-trip, instead of relying on the `PartialPeriod` itself.

Like I said, doing so is not straightforward on the `Manifest` structure. You're open to try it though.